sharding/handler/utils/shard_tracker_utils.py

```diff
@@ -1,23 +1,70 @@
 from eth_utils import (
+    event_abi_to_log_topic,
     to_dict,
+    to_checksum_address,
     decode_hex,
     big_endian_to_int,
 )
 
-from sharding.handler.utils.headers import (
-    CollationHeader,
+from sharding.contracts.utils.smc_utils import (
+    get_smc_json,
 )
 
 
 @to_dict
-def parse_collation_added_log(log):
-    # `shard_id` is the first indexed entry,hence the second entry in topics
-    shard_id_bytes32 = log['topics'][1]
+def parse_register_notary_log(log):
+    notary = log['topics'][1][-20:]
     data_bytes = decode_hex(log['data'])
-    header_bytes = shard_id_bytes32 + data_bytes[:-64]
-    is_new_head = bool(big_endian_to_int(data_bytes[-64:-32]))
-    score = big_endian_to_int(data_bytes[-32:])
-    collation_header = CollationHeader.from_bytes(header_bytes)
-    yield 'header', collation_header
-    yield 'is_new_head', is_new_head
-    yield 'score', score
+    index_in_notary_pool = big_endian_to_int(data_bytes[:32])
+    yield 'index_in_notary_pool', index_in_notary_pool
+    yield 'notary', to_checksum_address(notary)
```

Comment:
What do you think about pre-defining all the log format?  Using class or `namedtuple`?

Reply:
Sounds good. May even be better if these format can be extracted from contract abi in runtime. Working on it now.